File: 2020-py/_template_.py
import utils
from typing import List


def run_part1(title: str, input: List[str], correctResult: int):
    result = 0
    utils.validate_result(title, result, correctResult)


def run_part2(title: str, input: List[str], correctResult: int):
    result = 0
    utils.validate_result(title, result, correctResult)


if __name__ == "__main__":
    day = 0
    print(f"---- Day {day}: Title ----")

    run_part1("Test Case 1",
        utils.read_input_as_list(day, "example1"),
        0)
    # run_part1("problem",
    #     utils.read_input_as_list(day, "input"),
    #     0)

    # print("---- part 2 ----")

    # run_part2("Test Case 1",
    #     utils.read_input_as_list(day, "example1"),
    #     0)
    # run_part2("problem",
    #     utils.read_input_as_list(day, "input"),
    #     0)

File: 2019/CSharp/template.cs
namespace AdventOfCode.CSharp.Year2019;

/// <summary>
/// https://adventofcode.com/2019/day/17
/// </summary>
class Template : PuzzleBase
{
    const int DAY = 0;


    public override IEnumerable<string> SolvePuzzle()
    {
        yield return "Title";
        yield return string.Empty;

        yield return RunExample(() => " Ex. 1) " + RunPart1(GetPuzzleData(1, "example1")));
        yield return RunProblem(() => "Part 1) " + RunPart1(GetPuzzleData(1, "input")));

        yield return string.Empty;

        yield return RunExample(() => " Ex. 1) " + RunPart1(GetPuzzleData(2, "example1")));
        yield return RunProblem(() => "Part 2) " + RunPart1(GetPuzzleData(2, "input")));
    }


    class InputAnswer : InputAnswer<List<int>, int?> { }
    InputAnswer GetPuzzleData(int part, string name)
    {
        var result = new InputAnswer()
        {
            Input = InputHelper.LoadInputFile(DAY, name)
                .Select(l => l.ToInt32())
                .ToList(),
            ExpectedAnswer = InputHelper.LoadAnswerFile(DAY, part, name)?.FirstOrDefault()?.ToInt32()
        };

        return result;
    }


    string RunPart1(InputAnswer puzzleData)
    {
        var answer = 0;
        return Helper.GetPuzzleResultText($": {answer}", answer, puzzleData.ExpectedAnswer);
    }

    string RunPart2(InputAnswer puzzleData)
    {
        var answer = 0;
        return Helper.GetPuzzleResultText($": {answer}", answer, puzzleData.ExpectedAnswer);
    }

}
